### PR TITLE
Update React to 17.0.2

### PR DESCRIPTION
**Pull request by GuillaumeRx**

### CODE REVIEW

File: src/components/TokenMigration/components/TokenMigrationMultiTx.tsx.rej [hunk 1]
```diff
@@ -0,0 +1,20 @@
+diff a/src/components/TokenMigration/components/TokenMigrationMultiTx.tsx b/src/components/TokenMigration/components/TokenMigrationMultiTx.tsx	(rejected hunks)
```
Comment (FrederikBolding) [inline src/components/TokenMigration/components/TokenMigrationMultiTx.tsx.rej:1]: Whatever this is can be deleted.

File: src/components/__tests__/ContactLookupField.test.tsx [hunk 2]
```diff
@@ -132,8 +131,9 @@ describe('ContactLookupField', () => {
     const { container } = getComponent(getDefaultProps(), contacts, output);
     const input = container.querySelector('input');
     fireEvent.click(input!);
+    input!.focus();
```
Comment (FrederikBolding) [inline src/components/__tests__/ContactLookupField.test.tsx:134]: Why are these needed? They are added in multiple places.

File: src/features/Dashboard/components/WalletBreakdown/AccountDropdown.tsx.rej [hunk 3]
```diff
@@ -0,0 +1,5 @@
+diff a/src/features/Dashboard/components/WalletBreakdown/AccountDropdown.tsx b/src/features/Dashboard/components/WalletBreakdown/AccountDropdown.tsx	(rejected hunks)
```
Comment (FrederikBolding) [inline src/features/Dashboard/components/WalletBreakdown/AccountDropdown.tsx.rej:1]: Delete

File: src/features/PurchaseMembership/components/MembershipSelector.tsx [hunk 4]
```diff
@@ -68,7 +66,7 @@ export default function MembershipSelector({ name, value, onSelect }: Membership
       options={options}
       onChange={(option) => onSelect(option)}
       getOptionLabel={(option) => option.title}
-      optionComponent={({ data, selectOption }: OptionProps<IMembershipConfig>) => (
+      optionComponent={({ data, selectOption }: OptionProps<IMembershipConfig, false>) => (
```
Comment (FrederikBolding) [inline src/features/PurchaseMembership/components/MembershipSelector.tsx:69]: What is the extra flag here for?

File: package.json [hunk 5]
```diff
@@ -94,24 +94,30 @@
...
     "@babel/plugin-syntax-dynamic-import": "7.8.3",
-    "@babel/plugin-transform-modules-commonjs": "7.12.13",
-    "@babel/preset-env": "7.12.16",
-    "@babel/preset-react": "7.12.13",
+    "@babel/plugin-transform-modules-commonjs": "7.14.5",
+    "@babel/preset-env": "7.14.5",
+    "@babel/preset-react": "7.14.5",
     "@babel/preset-typescript": "7.12.16",
+    "@emotion/cache": "11.4.0",
+    "@emotion/core": "11.0.0",
```
Comment (FrederikBolding) [inline package.json:109]: Is this needed? 🤔

File: src/components/TranslateMarkdown.tsx [hunk 6]
```diff
@@ -11,11 +11,11 @@ interface Props {
 const TranslateMarkdown = ({ source }: Props) => {
   return (
     <Markdown
-      disallowedTypes={['html']}
-      renderers={{
-        root: React.Fragment,
-        link: (props) => <LinkApp isExternal={true} {...props} />,
-        paragraph: React.Fragment // Remove <p> added by react-markdown.
+      disallowedElements={['html']}
```
Comment (FrederikBolding) [inline src/components/TranslateMarkdown.tsx:14]: Has the API changed here? Why was it bumped?

File: src/components/TranslateMarkdown.tsx [hunk 7]
```diff
@@ -11,11 +11,11 @@ interface Props {
...
+      components={{
+        //@ts-expect-error bad typing on props
```
Comment (FrederikBolding) [inline src/components/TranslateMarkdown.tsx:16]: Is this fixable?

File: STRUCTURE.md [hunk 8]
```diff
@@ -27,23 +27,23 @@
 
 ## Providers
 
-* `Providers` are special React components that are used to provide functionality potentially many layers deep, while avoiding prop-drilling.
```
Comment (FrederikBolding) [inline STRUCTURE.md:30]: Why has the been changed? 😆

File: babel.config.js [hunk 9]
```diff
@@ -1,6 +1,11 @@
 module.exports = {
   presets: [
-    '@babel/preset-react',
+    [
+      '@babel/preset-react',
+      {
+        runtime: 'automatic'
```
Comment (FrederikBolding) [inline babel.config.js:6]: Why?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]
Left a few comments about things that I was curious about. Also needs a rebase

File: src/features/SendAssets/__tests__/SendAssets.test.tsx [hunk 10]
```diff
@@ -36,11 +34,12 @@ describe('SendAssetsFlow', () => {
 
   test('Can render too many decimals error.', async () => {
     const { getByText, container } = renderComponent();
-    const amount = container.querySelector('input[name="amount"]')!;
+    const amount = container.querySelector('input[name="amount"]')! as HTMLElement;
```
[the remark this answers is not in the document]
Comment (GuillaumeRx) [reply src/features/SendAssets/__tests__/SendAssets.test.tsx:37]: Yup it is, otherwise the needed prototypes are not available

File: src/components/__tests__/ContactLookupField.test.tsx (hunk 2 above)
Comment (GuillaumeRx) [reply src/components/__tests__/ContactLookupField.test.tsx:134]: Now with testing library when you want to type in a input you need to focus first, see https://testing-library.com/docs/guide-events/

File: src/components/TranslateMarkdown.tsx (hunk 6 above)
Comment (GuillaumeRx) [reply src/components/TranslateMarkdown.tsx:14]: Yup the API has changed and it was bumped to conform to the react version needed by the Library

File: STRUCTURE.md (hunk 8 above)
Comment (GuillaumeRx) [reply STRUCTURE.md:30]: Wait, I never touched this XD

File: babel.config.js (hunk 9 above)
Comment (GuillaumeRx) [reply babel.config.js:6]: This is required for the last react version :)

File: src/features/PurchaseMembership/components/MembershipSelector.tsx (hunk 4 above)
Comment (GuillaumeRx) [reply src/features/PurchaseMembership/components/MembershipSelector.tsx:69]: This is the `isMulti` flag. It's to define if the selector allow multiple selections

File: src/components/TranslateMarkdown.tsx (hunk 7 above)
Comment (GuillaumeRx) [reply src/components/TranslateMarkdown.tsx:16]: No because it doesn't provide the good props for the link

File: STRUCTURE.md (hunk 8 above)
Comment (GuillaumeRx) [reply STRUCTURE.md:30]: Oh yes I touched this to remove the `React` from `React.createContext()` (because we don't import React anymore) and it might have linted the file :)

File: package.json (hunk 5 above)
Comment (GuillaumeRx) [reply package.json]: After trying to remove them we still need them for the tests

**Comment by GuillaumeRx** [comment]
@FrederikBolding you're my man !

**Review by FrederikBolding — APPROVED** [review]

**Comment by Aeinsteroid54** [comment]
**Update React to 17.0.2#4043**
.patch

**Comment by Afsan751** [comment]
#4291 #4134 #4295 #2061 #4283
